scripts/Muon/GUI/Common/muon_workspace.py

```diff
@@ -0,0 +1,164 @@
+from mantid.api import Workspace, WorkspaceGroup
+
+
+def add_directory_structure(dirs):
+    """
+    create the nested WorkspaceGroup structure in the ADS specified by the
+    stored directory attribute.
+    dirs = ["dir1", "dir2"]
+    """
+    if not dirs:
+        return
+    if len(dirs) > len(set(dirs)):
+        raise ValueError("Group names must be unique")
+
+    for directory in dirs:
+        if not mtd.doesExist(directory):
+            workspace_group = api.WorkspaceGroup()
+            mtd.addOrReplace(directory, workspace_group)
+        elif not isinstance(mtd[directory], api.WorkspaceGroup):
+            mtd.remove(directory)
+            workspace_group = api.WorkspaceGroup()
+            mtd.addOrReplace(directory, workspace_group)
+        else:
+            # exists and is a workspace group
+            pass
+
+    # Create the nested group structure in the ADS
+    previous_dir = ""
+    for i, directory in enumerate(dirs):
+        if i == 0:
+            previous_dir = directory
+            continue
+        mtd[previous_dir].add(directory)
+        previous_dir = directory
+
+
+class MuonWorkspace(object):
+    """
+    A wrapper around a single workspace for use with MuonAnalysis.
+
+    A single instance of the wrapped workspace exists, but it is either in the ADS or stored in a
+    class attribute. Either way, a single property "workspace" will always return it; with the
+    show() and hide() methods storing/retrieving it from the the ADS. The show() attribute takes a
+    name, which is a string using a folder-like syntax to describe the workspaces location within a
+    nested structure of group workspaces
+
+    so for example name = "group1/group2/workspace" would store the workspace in the ADS inside
+    group2, which is inside group1. The "/" character is used to separate folders in the name as in
+    the example.
+
+    A basic muon workspace which is either the workspace or the name of the workspace in the ADS
+    """
+
+    def __init__(self, workspace):
+        self._is_in_ads = False
+        self._workspace = None
+        self._directory_structure = ""
+        self._workspace_name = ""
+
+        self.workspace = workspace
+
+    def __str__(self):
+        return "MuonWorkspace Object \n" \
+               "In ADS         : {}\n" \
+               "Name           : {}\n" \
+               "Directory      : {}\n" \
+               "Workspace Type : {}\n".format(self._is_in_ads,
+                                              self._workspace_name,
+                                              self._directory_structure,
+                                              type(self._workspace))
+
+    @property
+    def is_hidden(self):
+        """Is the workspace hidden (i.e. not in the ADS)."""
+        return not self._is_in_ads
+
+    @property
+    def name(self):
+        """The current name of the workspace."""
+        return self._directory_structure + self._workspace_name
```

Comment:
I would have thought name would just be the name of the workspace and not the name of the workspace including the 'path'